core/ts/Poll.ts

```diff
@@ -423,159 +428,201 @@ export class Poll implements IPoll {
-            newStateLeaf.voiceCreditBalance = BigInt(newStateLeaf.voiceCreditBalance.valueOf()) + BigInt(amount);
-            this.stateLeaves[Number(stateIndex)] = newStateLeaf;
-            this.stateTree.update(Number(stateIndex), newStateLeaf.hash());
-
-            // we still need them as placeholder for vote command
-            const currentBallot = this.ballots[Number(stateIndex)].copy();
-            currentBallots.unshift(currentBallot);
-            currentBallotsPathElements.unshift(this.ballotTree.genMerklePath(Number(stateIndex)).pathElements);
-            currentVoteWeights.unshift(currentBallot.votes[0]);
-
-            const vt = new IncrementalQuinTree(this.treeDepths.voteOptionTreeDepth, BigInt(0), 5, hash5);
-            for (let i = 0; i < this.ballots[0].votes.length; i++) {
-              vt.insert(currentBallot.votes[i]);
-            }
-
-            currentVoteWeightsPathElements.unshift(vt.genMerklePath(0).pathElements);
-          } catch (e) {
-            console.error("An error occurred: ", e.message);
-            throw e;
-          }
-          break;
-        default:
-          break;
-      } // end msgType switch
+            break;
+          default:
+            break;
+        }
+      } else {
+        // Since we don't have a command at that position, use a blank state leaf
+        currentStateLeaves.unshift(this.stateLeaves[0].copy());
+        currentStateLeavesPathElements.unshift(this.stateTree.genMerklePath(0).pathElements);
+        // since the command is invliad we use the blank ballot
+        currentBallots.unshift(this.ballots[0].copy());
+        currentBallotsPathElements.unshift(this.ballotTree.genMerklePath(0).pathElements);
+
+        // Since the command is invalid, we use a zero vote weight
+        currentVoteWeights.unshift(this.ballots[0].votes[0]);
+
+        // create a new quinary tree and add an empty vote
+        const vt = new IncrementalQuinTree(this.treeDepths.voteOptionTreeDepth, BigInt(0), STATE_TREE_ARITY, hash5);
+        vt.insert(this.ballots[0].votes[0]);
+
+        // get the path elements for this empty vote weight leaf
+        currentVoteWeightsPathElements.unshift(vt.genMerklePath(0).pathElements);
+      }
     }
 
-    // loop for batch
+    // store the data in the circuit inputs object
     circuitInputs.currentStateLeaves = currentStateLeaves.map((x) => x.asCircuitInputs());
     circuitInputs.currentStateLeavesPathElements = currentStateLeavesPathElements;
     circuitInputs.currentBallots = currentBallots.map((x) => x.asCircuitInputs());
     circuitInputs.currentBallotsPathElements = currentBallotsPathElements;
     circuitInputs.currentVoteWeights = currentVoteWeights;
     circuitInputs.currentVoteWeightsPathElements = currentVoteWeightsPathElements;
 
+    // record that we processed one batch
     this.numBatchesProcessed++;
 
     if (this.currentMessageBatchIndex > 0) {
       this.currentMessageBatchIndex -= batchSize;
```

Comment:
Do we need to prevent `this.currentMessageBatchIndex` from being negative? or it won't happen?

Reply:
we have this check before: `assert(this.currentMessageBatchIndex % batchSize === 0` -> so it means that it's either the same as `batchSize` or a multiple of it, thus when we do `this.currentMessageBatchIndex -= batchSize;` it can never be < 0